src/test/java/org/folio/services/ftp/SftpUploadServiceTest.java

```diff
@@ -0,0 +1,167 @@
+  }
+
+  @Test
+  public void testSuccessfulUpload(VertxTestContext vertxTestContext) throws Exception {
+    logger.info("=== Test successful upload ===");
+    Date end = new Date();
+    end.setTime(System.currentTimeMillis() - 864000000);
+
+    BatchVoucher batchVoucher = new BatchVoucher();
+    batchVoucher.setId(UUID.randomUUID().toString());
+    batchVoucher.setStart(end);
+    batchVoucher.setEnd(end);
+    batchVoucher.setBatchGroup(UUID.randomUUID().toString());
+    batchVoucher.setCreated(new Date());
+
+    SftpUploadService helper = new SftpUploadService(uri, 22);
+    var future = helper.upload(context, USERNAME, PASSWORD, EXPORT_FOLDER_NAME, FILENAME , JsonObject.mapFrom(batchVoucher).encodePrettily())
+      .onSuccess(logger::info)
+      .onFailure(t -> {
+        logger.error(t);
+        Assertions.fail(t.getMessage());
+      })
+      .onComplete(logger::info);
+    vertxTestContext.assertComplete(future)
+      .onSuccess(result -> vertxTestContext.completeNow());
+  }
+
+  @Test
+  void testSuccessfulUploadForLongPath(VertxTestContext vertxTestContext) throws Exception {
+    logger.info("=== Test successful upload ===");
+    Date end = new Date();
+    end.setTime(System.currentTimeMillis() - 864000000);
+
+    BatchVoucher batchVoucher = new BatchVoucher();
+    batchVoucher.setId(UUID.randomUUID().toString());
+    batchVoucher.setStart(end);
+    batchVoucher.setEnd(end);
+    batchVoucher.setBatchGroup(UUID.randomUUID().toString());
+    batchVoucher.setCreated(new Date());
+
+    SftpUploadService helper = new SftpUploadService(uri, 22);
+    var future = helper.upload(context, USERNAME, PASSWORD, EXPORT_FOLDER_NAME+"/test/long/path", FILENAME , JsonObject.mapFrom(batchVoucher).encodePrettily())
+      .onSuccess(logger::info)
+      .onFailure(t -> {
+        logger.error(t);
+        Assertions.fail(t.getMessage());
+      })
+      .onComplete(logger::info);
+    vertxTestContext.assertComplete(future)
+      .onSuccess(result -> vertxTestContext.completeNow());
+  }
+
+  @Test
+  void testFailedConnect(VertxTestContext vertxTestContext) throws Exception {
+    logger.info("=== Test failed connect ===");
+    Date end = new Date();
+    end.setTime(System.currentTimeMillis() - 864000000);
+    BatchVoucher batchVoucher = new BatchVoucher();
+
+    SftpUploadService helper = new SftpUploadService(INVALID_URI, 22);
+    var future = helper.upload(context, USERNAME, PASSWORD, EXPORT_FOLDER_NAME+"/test/long/path", FILENAME , JsonObject.mapFrom(batchVoucher).encodePrettily())
+      .onFailure(logger::info)
+      .onComplete(logger::info);
+    vertxTestContext.assertFailure(future)
+      .onComplete(result -> vertxTestContext.completeNow());
+  }
+
+  @Test
+  void testSuccessfulDownload() throws Exception {
+    SftpClient sftpClient = getSftpClient(USERNAME, PASSWORD, sftp.getHost(), sftp.getMappedPort(22));
+    byte[] fileBytes = download(sftpClient, EXPORT_FOLDER_NAME + "/" + FILENAME);
```

Comment:
You can create new string from this byte[] array and compare that file has the same content as you expect

Reply:
added checks

Reply:
can you please move this check to testSuccessfulUpload, because we should make our unit tests isolated

Reply:
and add this check to all cases for successful uploads

Reply:
added